pkgs/os-specific/linux/trusting-trust/mes/default.nix

```diff
@@ -0,0 +1,297 @@
+    mescc ''${MES_PREFIX}/lib/stub/localtime.c
+    mescc ''${MES_PREFIX}/lib/stub/sigemptyset.c
+    mescc ''${MES_PREFIX}/lib/x86-mes-mescc/setjmp.c
+    mescc ''${MES_PREFIX}/lib/linux/close.c
+    mescc ''${MES_PREFIX}/lib/linux/rmdir.c
+    mescc ''${MES_PREFIX}/lib/linux/stat.c
+    catm libc+tcc.a libc.a islower.o isupper.o tolower.o toupper.o abtod.o dtoab.o search-path.o execvp.o fclose.o fdopen.o ferror.o fflush.o fopen.o fprintf.o fread.o fseek.o ftell.o fwrite.o printf.o remove.o snprintf.o sprintf.o sscanf.o vfprintf.o vprintf.o vsnprintf.o vsprintf.o vsscanf.o calloc.o qsort.o strtod.o strtof.o strtol.o strtold.o strtoll.o strtoul.o strtoull.o memmem.o strcat.o strchr.o strlwr.o strncpy.o strrchr.o strstr.o strupr.o sigaction.o ldexp.o mprotect.o localtime.o sigemptyset.o setjmp.o close.o rmdir.o stat.o
+    catm libc+tcc.s libc.s islower.s isupper.s tolower.s toupper.s abtod.s dtoab.s search-path.s execvp.s fclose.s fdopen.s ferror.s fflush.s fopen.s fprintf.s fread.s fseek.s ftell.s fwrite.s printf.s remove.s snprintf.s sprintf.s sscanf.s vfprintf.s vprintf.s vsnprintf.s vsprintf.s vsscanf.s calloc.s qsort.s strtod.s strtof.s strtol.s strtold.s strtoll.s strtoul.s strtoull.s memmem.s strcat.s strchr.s strlwr.s strncpy.s strrchr.s strstr.s strupr.s sigaction.s ldexp.s mprotect.s localtime.s sigemptyset.s setjmp.s close.s rmdir.s stat.s
+
+    # Build mes itself
+    mescc ''${MES_PREFIX}/src/builtins.c
+    mescc ''${MES_PREFIX}/src/cc.c
+    mescc ''${MES_PREFIX}/src/core.c
+    mescc ''${MES_PREFIX}/src/display.c
+    mescc ''${MES_PREFIX}/src/eval-apply.c
+    mescc ''${MES_PREFIX}/src/gc.c
+    mescc ''${MES_PREFIX}/src/globals.c
+    mescc ''${MES_PREFIX}/src/hash.c
+    mescc ''${MES_PREFIX}/src/lib.c
+    mescc ''${MES_PREFIX}/src/math.c
+    mescc ''${MES_PREFIX}/src/mes.c
+    mescc ''${MES_PREFIX}/src/module.c
+    mescc ''${MES_PREFIX}/src/posix.c
+    mescc ''${MES_PREFIX}/src/reader.c
+    mescc ''${MES_PREFIX}/src/stack.c
+    mescc ''${MES_PREFIX}/src/string.c
+    mescc ''${MES_PREFIX}/src/struct.c
+    mescc ''${MES_PREFIX}/src/symbol.c
+    mescc ''${MES_PREFIX}/src/vector.c
+
+    # Install libraries
+    cp libc.a ''${MES_PREFIX}/lib/x86-mes/
+    cp libc+tcc.a ''${MES_PREFIX}/lib/x86-mes/
+    cp libmescc.a ''${MES_PREFIX}/lib/x86-mes/
+    cp libc.s ''${MES_PREFIX}/lib/x86-mes/
+    cp libc+tcc.s ''${MES_PREFIX}/lib/x86-mes/
+    cp libmescc.s ''${MES_PREFIX}/lib/x86-mes/
+    cp crt1.o ''${MES_PREFIX}/lib/x86-mes/
+    cp crt1.s ''${MES_PREFIX}/lib/x86-mes/
+
+    # Link everything into new mes executable
+    ''${out}/bin/mes-m2 -e main ''${out}/bin/mescc.scm -- --base-address 0x08048000 -L ''${MES_PREFIX}/lib -nostdlib -o ''${out}/bin/mes -L . crt1.o builtins.o cc.o core.o display.o eval-apply.o gc.o globals.o hash.o lib.o math.o mes.o module.o posix.o reader.o stack.o string.o struct.o symbol.o vector.o -lc -lmescc
+  '';
+}) // {
+  mesPrefix = "/share/mes-${version}";
+}
```

Comment:
Would be good if derivations for packages like this had the standard meta attributes.